Rakefile

```diff
@@ -30,6 +40,14 @@ task :generate_test_gem => ['engine_cart:setup'] do
   IO.write(".internal_test_gem/internal_test_gem.gemspec", File.open(".internal_test_gem/internal_test_gem.gemspec") {|f| f.read.gsub(/.*homepage.*/, "")})
   IO.write(".internal_test_gem/internal_test_gem.gemspec", File.open(".internal_test_gem/internal_test_gem.gemspec") {|f| f.read.gsub(/.*sqlite3.*/, "")})
 
+  if Rails.version.to_f < 6.0 || Rails.version.to_f == 6.0
+    gemspec_file_path = ".internal_test_gem/internal_test_gem.gemspec"
+    IO.write(gemspec_file_path,
+             File.open(gemspec_file_path) { |f|
+                f.read.gsub(/(spec.add_dependency "rails".+$)/, "\\1\n  spec.add_dependency 'sprockets', '~> 3.7'")
```

Comment:
I'm hoping that this will no longer be necessary with https://github.com/rails/rails/issues/36954#issuecomment-522171807 in 6.0.1, as I suspect these are errors related to using 4.0.0 as the `sprockets` dependency.